### ISSUE TITLE
Android port

### ISSUE THREAD

**Issue by ykonstant1** [ahrm/sioyek#840]
I read in an earlier issue that you are planning to port Sioyek to Android and iOS devices.  I find that a very exciting prospect, because I use an android tablet to read papers and having all the niceties of Sioyek (especially the overview!) would be amazing. I would like to know if there is any progress on that port, and/or what the major pain points are.

Thanks!

**Comment by ykonstant1** [comment]
Understandable; I have heard horror stories about iOS development.  This is very exciting news nevertheless, and I am looking forward to the ports!

**Comment by HakonHarnes** [comment]
+1 Also very exited for this! Didn't know you were also working on an iOS version. Are you planning on synchronizing highlights and bookmarks between the desktop/mobile apps or will it be kept locally only?

**Comment by ahrm** [comment]
In the android version it is possible to synchronize the database using dropbox and and an app called `FolderSync`. I don't know if it is possible on iOS and even on android it is pretty inconvenient. I am thinking about implementing a specialized server-side syncing mechanism but if I do it will be a paid subscription service because of server upkeep costs. I am curious if there is any demand for that at all?

**Comment by HakonHarnes** [comment]
> I am curious if there is any demand for that at all?

Yes. When reading papers for my master's thesis I used to read papers on my iPad and annotate them there, then import them on my computer later.  I tried the `embed_annotations` command but it didn't work that well for me. If I were to read it again on the iPad I would have to embed the annotations from Sioyek, add more annotations on the iPad, export it again, and re-extract the highlights. Quite the hazzle with syncing between devices! 

> In the android version it is possible to synchronize the database using dropbox and and an app called FolderSync. I don't know if it is possible on iOS and even on android it is pretty inconvenient.

An  iOS I use called "HeavySet" does it in a similar way - but it uses Google Drive or iCloud I believe. 

> I am thinking about implementing a specialized server-side syncing mechanism but if I do it will be a paid subscription service because of server upkeep costs.

This would be the preferred way but more complex of course. You'd also need some sort of authentication and authorization mechanisms. I would recommend looking into Amazon Web Services (AWS) if you go this route. They have some fairly reasonable (always) free tiers, like 25 GB of storage with DynamoDB and 1 million AWS lambda function calls. If you exceed 25 GB you will of course have to pay, but you'll only pay for what you use. 

I would lean against not doing server-side syncing as it will complicate stuff a lot, and I'm unsure how many users actually need cross-device syncing. To be fair, I don't even need it at the moment - I was just curious.

**Comment by zetashift** [comment]
> In the android version it is possible to synchronize the database using dropbox and and an app called `FolderSync`. I don't know if it is possible on iOS and even on android it is pretty inconvenient. I am thinking about implementing a specialized server-side syncing mechanism but if I do it will be a paid subscription service because of server upkeep costs. I am curious if there is any demand for that at all?

Hopefully it's possible with Syncthing as well! I use syncthing to sync my notes and it's been pretty painless.
A subscription service sounds like a fine solution for people not wanting to do the manual work themselves.

**Comment by PilgrimRy** [comment]
This would be amazing. Geez, i wish i was a more capable programmer and could contribute.

**Comment by ouboub** [comment]
Hi
are there any news related to the iOS release?
Since I use my iPad for teaching and research very much, that would be a game changer!
Ok, I will buy you a coffee first  :)
Uwe Brauer

**Comment by ahrm** [comment]
Hi, we I do have some progress on the iOS port but I don't want to give a timeline because it probably will be wrong. It definitely will come out at some point though.

**Comment by stradichenko** [comment]
Hi, I'm curious on this feature, how is the Android/IOS version going? If it's done, are there special instructions for installation?
Also; do you think is it possible to run customs scripts as in the PC version?

**Comment by ahrm** [comment]
The development branch of this repo can be compiled for android (no binary releases yet, you'll have to compile it from source if you want). It definitely will not be possible to run custom scripts though (I am pretty sure apple does not allow apps that can be scripted, also I don't see how it will be possible on android).

**Comment by niv-l** [comment]
Hi, I am interested in compiling the Android version. I am not too familiar with how building for mobile looks like, what commands should I run?

**Comment by ahrm** [comment]
I don't remember exactly, I don't think it is much different from building it for other platforms (at least using qt creator). The main thing is you need to compile libmupdf for android, which you can refer to mupdf documentation for that.

**Comment by niv-l** [comment]
Thank you! I'll try to attempt a build; if it goes well, I'll briefly mention the steps I took for those interested.

**Comment by pankajbhanu** [comment]
@niv-l Hi, were you able to build the android project from development branch? If yes, could you pls post the steps here.

**Comment by srfsh** [comment]
If anyone knows how to build it for android using the cli, it would be highly appreciated if they share the instructions here.

**Comment by RasmusMathiesen** [comment]
Being able to run sioyek on an android e-ink tablet would be incredible. iOS support seems to be less requested than android, so maybe it could make sense to release android builds even though an iOS version is not ready yet? Crossing my fingers, thanks either way for the amazing work.

**Comment by k10876** [comment]
> If anyone knows how to build it for android using the cli, it would be highly appreciated if they share the instructions here.

I actually did build them, fortunately with the AI things that are around here. However very unfortunately the **keyboards don't work** although the keys went in, which shows that this thing is actually half complete. 

I'll try to see if some magic happens with some patches and share more info here.

**Comment by k10876** [comment]
Hi guys,

I managed to build something at https://github.com/k10876/sioyek/actions/runs/26242862259.

On my tablet (Samsung Galaxy Tab S8+) it succeeded to run and parse keyboards (which is not the case for the original build). **This is a arm64-v8a debug build and is signed by debug signature!** 

Please note that this is an **AI-driven** build and is nowhere near stable (however it's *usable*, just for now). For some unknown reason, it's 1000+ commits behind the dev branch, and I'll try to merge it if I have more time.

Not all github actions artifacts will work, since i'll be making crazy changes and put it live. Be careful for that.
